src/main/java/tools/jackson/databind/deser/bean/BeanDeserializer.java

```diff
@@ -915,6 +919,15 @@ protected Object deserializeWithUnwrapped(JsonParser p, DeserializationContext c
                 handleIgnoredProperty(p, ctxt, bean, propName);
                 continue;
             }
+
+            // [databind#1709]: Track non-null unwrapped property
+            if (unwrappedPropertyNames != null
+                    && propName != null && unwrappedPropertyNames.contains(propName)) {
+                if (p.currentToken() != JsonToken.VALUE_NULL) {
```

Comment:
I don't think this check is needed; all we need to know if there is at least one property of any value (even `null`) -- and then need to specifically construct "empty" value.

So instead of `hasNonNullUnwrappedContent` just `hasAnyUnwrappedContent`